tests-legacy/Unit/Classes/ToolsCoreTest.php

```diff
@@ -369,6 +369,48 @@ public function testStrReplaceFirst($search, $replace, $subject, $cur, $expected
         $this->assertEquals($expected, Tools::StrReplaceFirst($search, $replace, $subject, $cur));
     }
 
+    /**
+     * @param string $passwordGenerated
+     * @param string $expectedPassword
+     *
+     * @dataProvider passwodGenProvider
+     */
+    public function testPasswdGen($expectedPassword, $passwordGenerated)
+    {
+        $this->assertRegExp($expectedPassword, $passwordGenerated, 'The password generated ' . $expectedPassword . ' no match with ' . $expectedPassword);
```

Comment:
Hi @PululuK the use of `assertRegExp` can be powerful but it might not be very readable 😅  . In order to understand the test, the reader must decipher the input regexp to find out what you are trying to validate.

Could we find something that makes the test easier to read?

Readability is quite important for test because to maintain something you need to understand it. If it is hard to understand it is hard to maintain, and hard to maintain tests lose usefulness

Reply:
@matks  can you please tell me what does not seem readable to you here?

Reply:
```
'/^(\w){10}$/' => Tools::passwdGen(10, Tools::PASSWORDGEN_FLAG_ALPHANUMERIC),
'/^(false|0)$/' => Tools::passwdGen(1),
```

These 2 statements are not obvious. When I read them first I was "gné 😮 ?"

Then I saw it was regexp, then I understood I had to convert it in my mind into something meaningful and finally I got it. However it means to read a single line took me 10 seconds.

It's not "unreadable" 😉  (this is why I added a comment but I dit not `request changes`). What I'm saying is "it could be a lot more readable"

I have some ideas to improve readability but I prefer to let you choose what is a suitable solution. [Rather than giving explicit directions I prefer to let the developer decide.](https://google.github.io/eng-practices/review/reviewer/comments.html) following Google Code Review Practices 😄  .

Reply:
> I have some ideas to improve readability but I prefer to let you choose what is a suitable solution.

it is indeed very strange to hear in an exercise of code reviews.

Reply:
I just want you to understand what I am going to write are _suggestions_ on how to make the code more readable, but you can _discard these suggestions_ if you find something else, than you prefer, that also improve the readability

```
self::TEN_ALPHANUMERIC => Tools::passwdGen(10, Tools::PASSWORDGEN_FLAG_ALPHANUMERIC),
self::FALSE_OR_ZERO => Tools::passwdGen(1),
```

This would be one way to make the lines easier to read 😄  . Just put a name (here a constant but it could be a variable) on the regexp. So the reader can grasp what is this line without having to decipher the regexp.

Reply:
your proposal seems very interesting, luckily you haven't kept it to yourself !
Thanks @matks